src/Admin/AbstractAdmin.php

```diff
@@ -780,35 +745,27 @@ public function getFilterParameters()
      * value (ie the parent object) or to filter the object.
      *
      * @throws \InvalidArgumentException
-     *
-     * @return string|null
      */
-    public function getParentAssociationMapping()
+    public function getParentAssociationMapping(): ?string
     {
-        // NEXT_MAJOR: remove array check
-        if (\is_array($this->parentAssociationMapping) && $this->getParent()) {
-            $parent = $this->getParent()->getCode();
+        if (!$this->getParent()) {
+            return null;
+        }
 
-            if (\array_key_exists($parent, $this->parentAssociationMapping)) {
-                return $this->parentAssociationMapping[$parent];
-            }
+        $parent = $this->getParent()->getCode();
 
-            throw new \InvalidArgumentException(sprintf(
-                "There's no association between %s and %s.",
-                $this->getCode(),
-                $this->getParent()->getCode()
-            ));
+        if (\array_key_exists($parent, $this->parentAssociationMapping)) {
+            return $this->parentAssociationMapping[$parent];
         }
 
-        // NEXT_MAJOR: remove this line
-        return $this->parentAssociationMapping;
+        throw new \InvalidArgumentException(sprintf(
+            "There's no association between %s and %s.",
```

Comment:
IMO (in the scope of other PR) `\InvalidArgumentException` should be replaced by a proper exception (like `\DomainException` or similar), since `$this->getCode()` and `$this->getParent()->getCode()` aren't arguments.

Reply:
I'm 👍for custom exceptions. 

In this particular case I have to check, but maybe this can be fixed with something like this:
```php
$child->setParent($this);
$child->addParentAssociationMapping($this->getCode(), $parentMappingField);
// Replace to
$child->setParent($this, $parentMappingField);
```